oci/model.py

```diff
@@ -43,8 +43,24 @@ class OciRegistryType(enum.Enum):
     GCR = 'gcr'
     DOCKERHUB = 'dockerhub'
     ARTIFACTORY = 'artifactory'
+    GHCR = 'ghcr'
+    ALIBABA = 'alibaba'
     UNKNOWN = 'unknown'
 
+    @staticmethod
+    def from_image_ref(image_reference: str) -> 'OciRegistryType':
+        if 'pkg.dev' in image_reference:
```

Comment:
nit: how about using OciImageReference + its `netloc` property to avoid "false-positives"?